src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1682,7 +1688,27 @@ static bool processSubscriptions
     }
 
     /* Check 2: expression (q) */
-    if (!matchExpression(notifyCerP, trigs->expression.q))
+    if (noCache == true)
```

Comment:
The idea behin TriggeredSubscriptions is to have there all the information that, at processSubscription() time, the logic will need to send notifications. Thus, I think that to be coherent with that idea, the parsing should be done at that time.